server/mail.ts

```diff
@@ -14,6 +14,22 @@ export type Message = {
 
 type SendEmailFn = (msg: Message & { from: string }, config: gitfs.Config) => Promise<string>
 
+export const validateMessage = (msg: Message) => {
+    // simple regex: at least one character before the @, before the period and after it
+    const emailRegex = /^.+@.+\..+$/
+    const errors = []
+
+    if(msg.from && (!emailRegex.test(msg.from) || msg.subject.length > 254)) errors.push("sender")
+    if(!emailRegex.test(msg.to) || msg.subject.length > 254) errors.push("recipient");
+    if(typeof msg.subject !== "string" || msg.subject.length > 255) errors.push("subject")
+    if(typeof msg.text !== "string") errors.push("content")
+
+    if(errors.length) {
+        return "email message validation failed because of fields: " + errors.join(", ")
+    }
+    return null
```

Comment:
The validation function should return a json object with *only* the fields it has validated. Otherwise, the remote can pass arbitrary json fields, which might be interpreted by sendgrid/mailgun.